src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -445,9 +439,6 @@ private int getAndSetNetworkTimeout(final Connection connection, final long time
                if (validationTimeout < SECONDS.toMillis(1)) {
                   LOGGER.warn("{} - A validationTimeout of less than 1 second cannot be honored on drivers without setNetworkTimeout() support.", poolName);
                }
-               else if (validationTimeout % SECONDS.toMillis(1) != 0) {
-                  LOGGER.warn("{} - A validationTimeout with fractional second granularity cannot be honored on drivers without setNetworkTimeout() support.", poolName);
-               }
```

Comment:
This is lack of (reading) documentation that leaks in source with checks and messages.